client/data/settings/hooks.js

```diff
@@ -0,0 +1,64 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from '../constants';
+
+export const useSettings = () => {
+	const { saveSettings } = useDispatch( STORE_NAME );
+
+	return useSelect(
+		( select ) => {
+			const {
+				getSettings,
+				hasFinishedResolution,
+				isResolving,
+				isSavingSettings,
+			} = select( STORE_NAME );
+
+			const isLoading =
+				isResolving( 'getSettings' ) ||
+				! hasFinishedResolution( 'getSettings' );
+
+			return {
+				settings: getSettings(),
+				isLoading,
+				saveSettings,
+				isSaving: isSavingSettings(),
+			};
+		},
+		[ saveSettings ]
+	);
```

Comment:
Same as below
```suggestion
	const settings = useSelect( ( select ) => {
		const { getSettings } = select( STORE_NAME );

		return getSettings();
	}, [] );

	const isLoading = useSelect( ( select ) => {
		const { hasFinishedResolution, isResolving } = select( STORE_NAME );

		return (
			isResolving( 'getSettings' ) ||
			! hasFinishedResolution( 'getSettings' )
		);
	}, [] );

	const isSaving = useSelect( ( select ) => {
		const { isSavingSettings } = select( STORE_NAME );

		return isSavingSettings();
	}, [] );

	return { settings, isLoading, isSaving, saveSettings };
};
```

Reply:
That's brilliant! I give this change 5/5 rockets: :rocket: :rocket: :rocket: :rocket: :rocket: